public/app/features/dashboard/dashgrid/PanelChrome.tsx

```diff
@@ -118,7 +119,15 @@ export class PanelChrome extends PureComponent<Props, State> {
         }
       }
 
-      this.setState({ isFirstLoad, errorMessage, data });
+      const stateUpdate = { isFirstLoad, errorMessage, data };
+
+      if (this.isVisible) {
+        this.setState(stateUpdate);
+      } else {
+        // if we are getting data while another panel is in fullscreen / edit mode
+        // we need to store the data but not update state yet
+        this.delayedStateUpdate = stateUpdate;
+      }
```

Comment:
After #15554, we could consider adding something like:
panel.getQueryRunner().enableStreamEvents( this.props.isInView ) when `isInView` changes:
https://github.com/grafana/grafana/pull/15554/files#diff-791865e8ee91eb8f3e2cfa67eff71ce5R96

Reply:
what if the panel query runner is shared with another panel that is visible? :)

Reply:
better idea!

We just `unsubscribe()` when isInView becomes false, and re-subscribe when isInView becomes true.  `subscribe()` immediately updates the state with any changes that happened while disconnected.

As implemented, the streams stay open even if everyone has disconnected.  They only get closed if the next query/refresh does not match any open streams.

Lets wait till #15554 for this change since that PR makes visibility part of the state :)